### DISCUSSION TITLE
GraphQL Query Depth Limiting

### DISCUSSION THREAD

**Discussion by 4815162342** [graphql/graphql-http#50] [Q&A] [answered]
Hey everyone - like a few others, I'm migrating my express GraphQL servers from the `express-graphql` package to this. One thing I really enjoyed was the drop-in package `graphql-depth-limit` that allowed for quick validation against unbounded query depths. 

Is anyone aware of a similar solution for this package? Or an easy way to adapt that one for `graphql-http`? Thanks in advance!

**Comment by enisdenjo** [comment, marked as answer]
Hey there, I've just released [v1.14.0](https://github.com/graphql/graphql-http/releases/tag/v1.14.0) with #51 that simplifies this process. You can add the depth limit validation rule from the `graphql-depth-limit ` package like this:

```js
import { createHandler } from 'graphql-http';
import depthLimit from 'graphql-depth-limit';
import { schema } from './my-schema';

const handler = createHandler({
  schema,
  validationRules: [depthLimit(7)],
});
```

**Reply by 4815162342** [reply]
@enisdenjo thank you so much! That's a huge help. 

And thanks overall for maintaining this open source package! It's devs like you who keep the industry moving forward :)
